manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -74,15 +75,21 @@ func (tr *TaskReaper) Run() {
 					serviceID: t.ServiceID,
 					nodeID:    t.NodeID,
 				}] = struct{}{}
-				if len(tr.dirty) > maxDirty {
-					timer.Stop()
-					tr.tick()
-				} else {
-					timer.Reset(reaperBatchingInterval)
+			case state.EventUpdateTask:
+				t := v.Task
+				if t.Status.State >= api.TaskStateOrphaned {
+					tr.orphaned = append(tr.orphaned, t.ID)
```

Comment:
If we follow the pattern that the task reaper deletes `ORPHANED` tasks immediately, then `Run` should begin by listing existing `ORPHANED` tasks and adding those to `tr.orphaned`. Though personally I still don't see the utility of adding this state if all it does is cause a task to be deleted immediately. I understand that in the future we may want to keep some `ORPHANED` tasks around for recordkeeping, but I don't see a reason to add the state in advance of that.